Update dependency env_logger from 0.6 to 0.10.2

jimmystewpot opened the pull request:
The `env_logger` API has changed in recent revisions, bringing up the dependency to a more recent version to make the codebase easier to maintain in the long term.

A test that defaults to ignored has been left to test any future API changes if required; I have left it there just for simple iterative testing, although it could be removed as it's trivial. Otherwise, this shouldn't break anything.

tomhoule reviewed (approved):
very nice, thanks for the PR!

tomhoule commented:
Ci is failing with

```

the package `graphql_client_cli` depends on `env_logger`, with features: `anstream` but `env_logger` does not have these features.
 It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.


```

I don't have the bandwidth to check right now, but it should be straightforward to fix.

jimmystewpot commented:
@tomhoule, thanks for looking at this pr so quickly. I have not been checking the MSRV information on packages, so this one slipped through the net.

The problem with CI is caused by the minimum version for graphql-client 1.64 vs. a style (via env_logger), which is 1.71. I reverted the env_logger to 0.10.2, which is the last version that supports 1.64. I started working on this update to make cross-platform terminal colours less buggy. I've now reverted the code back to pre-change and left only the dependency update in place (to 0.10.2)

Should the minimum version of rust supported by graphql-client by bumped to something more recent? (This is a supplementary question)

In Summary, the version has gone from 0.6 to 0.10.2 for env_logger; I've updated the subject/title of the PR to reflect the difference. I've also removed the tests.

tomhoule commented:
> Should the minimum version of rust supported by graphql-client by bumped to something more recent? (This is a supplementary question)

Absolutely. I'm going to cut a release very soon, and we can make a few breaking changes, including bumping the MSRV right after that.
